### ISSUE TITLE
gt sketch Assertion failed while generating gene model image

### ISSUE THREAD

**Issue by Brett-CCG** [genometools/genometools#527]
Hi, I received the following error when running gt sketch:
```
ccgstaff@brlscratch:~/bchapman/Grape/Chapter4_figures/chr18$ gt sketch -start 4109200 -end 4112700 -style default.style -force gene.png 4.AUGUSTUS_PREDICTION3
warning: seqid "chr18" on line 2 in file "4.AUGUSTUS_PREDICTION3" has not been previously introduced with a "##sequence-region" line, create such a line automatically
Assertion failed: (gt_hashmap_get(d->nodeinfo, rep)), function process_node, file src/annotationsketch/diagram.c, line 621.
This is a bug, please report it.
```
I was generating an illustration of the gene models reported by Augustus, and received the error.

Heres an example of a particular gene model I'm interested in. However, I received the error with the whole GFF file for this chromosome, as well as just this GFF of the gene model.
```
chr18   AUGUSTUS        gene    4098351 4112761 .       -       .       ID=g68938
chr18   AUGUSTUS        mRNA    4098351 4112761 0.69    -       .       ID=g68938.t1;Parent=g68938
chr18   AUGUSTUS        CDS     4098351 4098407 1       -       0       ID=g68938.t1.cds;Parent=g68938.t1
chr18   AUGUSTUS        intron  4098408 4098489 1       -       .       Parent=g68938.t1
chr18   AUGUSTUS        CDS     4098490 4100643 1       -       0       ID=g68938.t1.cds;Parent=g68938.t1
chr18   AUGUSTUS        intron  4100644 4100780 1       -       .       Parent=g68938.t1
chr18   AUGUSTUS        CDS     4100781 4100957 1       -       0       ID=g68938.t1.cds;Parent=g68938.t1
chr18   AUGUSTUS        intron  4100958 4101153 1       -       .       Parent=g68938.t1
chr18   AUGUSTUS        CDS     4101154 4101367 1       -       1       ID=g68938.t1.cds;Parent=g68938.t1
chr18   AUGUSTUS        intron  4101368 4101455 1       -       .       Parent=g68938.t1
chr18   AUGUSTUS        CDS     4101456 4101805 1       -       0       ID=g68938.t1.cds;Parent=g68938.t1
chr18   AUGUSTUS        intron  4101806 4101946 1       -       .       Parent=g68938.t1
chr18   AUGUSTUS        CDS     4101947 4102184 1       -       1       ID=g68938.t1.cds;Parent=g68938.t1
chr18   AUGUSTUS        intron  4102185 4102577 1       -       .       Parent=g68938.t1
chr18   AUGUSTUS        CDS     4102578 4102918 1       -       0       ID=g68938.t1.cds;Parent=g68938.t1
chr18   AUGUSTUS        intron  4102919 4103005 1       -       .       Parent=g68938.t1
chr18   AUGUSTUS        CDS     4103006 4103275 1       -       0       ID=g68938.t1.cds;Parent=g68938.t1
chr18   AUGUSTUS        intron  4103276 4103525 1       -       .       Parent=g68938.t1
chr18   AUGUSTUS        CDS     4103526 4103648 1       -       0       ID=g68938.t1.cds;Parent=g68938.t1
chr18   AUGUSTUS        intron  4103649 4103766 1       -       .       Parent=g68938.t1
chr18   AUGUSTUS        CDS     4103767 4103844 1       -       0       ID=g68938.t1.cds;Parent=g68938.t1
chr18   AUGUSTUS        intron  4103845 4104429 1       -       .       Parent=g68938.t1
chr18   AUGUSTUS        CDS     4104430 4104606 1       -       0       ID=g68938.t1.cds;Parent=g68938.t1
chr18   AUGUSTUS        intron  4104607 4104688 1       -       .       Parent=g68938.t1
chr18   AUGUSTUS        CDS     4104689 4104775 1       -       0       ID=g68938.t1.cds;Parent=g68938.t1
chr18   AUGUSTUS        intron  4104776 4104879 1       -       .       Parent=g68938.t1
chr18   AUGUSTUS        CDS     4104880 4104990 1       -       0       ID=g68938.t1.cds;Parent=g68938.t1
chr18   AUGUSTUS        intron  4104991 4105178 1       -       .       Parent=g68938.t1
chr18   AUGUSTUS        CDS     4105179 4105367 1       -       0       ID=g68938.t1.cds;Parent=g68938.t1
chr18   AUGUSTUS        intron  4105368 4105440 1       -       .       Parent=g68938.t1
chr18   AUGUSTUS        CDS     4105441 4105632 1       -       0       ID=g68938.t1.cds;Parent=g68938.t1
chr18   AUGUSTUS        intron  4105633 4105759 1       -       .       Parent=g68938.t1
chr18   AUGUSTUS        CDS     4105760 4105846 1       -       0       ID=g68938.t1.cds;Parent=g68938.t1
chr18   AUGUSTUS        intron  4105847 4106562 0.98    -       .       Parent=g68938.t1
chr18   AUGUSTUS        CDS     4106563 4106667 1       -       0       ID=g68938.t1.cds;Parent=g68938.t1
chr18   AUGUSTUS        intron  4106668 4106800 1       -       .       Parent=g68938.t1
chr18   AUGUSTUS        CDS     4106801 4106857 1       -       0       ID=g68938.t1.cds;Parent=g68938.t1
chr18   AUGUSTUS        intron  4106858 4106959 1       -       .       Parent=g68938.t1
chr18   AUGUSTUS        CDS     4106960 4107034 1       -       0       ID=g68938.t1.cds;Parent=g68938.t1
chr18   AUGUSTUS        intron  4107035 4107150 1       -       .       Parent=g68938.t1
chr18   AUGUSTUS        CDS     4107151 4107267 1       -       0       ID=g68938.t1.cds;Parent=g68938.t1
chr18   AUGUSTUS        intron  4107268 4107370 1       -       .       Parent=g68938.t1
chr18   AUGUSTUS        CDS     4107371 4107495 1       -       2       ID=g68938.t1.cds;Parent=g68938.t1
chr18   AUGUSTUS        intron  4107496 4108928 0.89    -       .       Parent=g68938.t1
chr18   AUGUSTUS        CDS     4108929 4108968 1       -       0       ID=g68938.t1.cds;Parent=g68938.t1
chr18   AUGUSTUS        intron  4108969 4109062 1       -       .       Parent=g68938.t1
chr18   AUGUSTUS        CDS     4109063 4109113 1       -       0       ID=g68938.t1.cds;Parent=g68938.t1
chr18   AUGUSTUS        intron  4109114 4109204 1       -       .       Parent=g68938.t1
chr18   AUGUSTUS        CDS     4109205 4109392 1       -       2       ID=g68938.t1.cds;Parent=g68938.t1
chr18   AUGUSTUS        intron  4109393 4109497 1       -       .       Parent=g68938.t1
chr18   AUGUSTUS        CDS     4109498 4109591 1       -       0       ID=g68938.t1.cds;Parent=g68938.t1
chr18   AUGUSTUS        intron  4109592 4110516 1       -       .       Parent=g68938.t1
chr18   AUGUSTUS        CDS     4110517 4110690 1       -       0       ID=g68938.t1.cds;Parent=g68938.t1
chr18   AUGUSTUS        intron  4110691 4110881 1       -       .       Parent=g68938.t1
chr18   AUGUSTUS        CDS     4110882 4110977 1       -       0       ID=g68938.t1.cds;Parent=g68938.t1
chr18   AUGUSTUS        intron  4110978 4111082 1       -       .       Parent=g68938.t1
chr18   AUGUSTUS        CDS     4111083 4111175 1       -       0       ID=g68938.t1.cds;Parent=g68938.t1
```
Thanks for any help you can provide.

Brett

**Comment by satta** [comment]
Hi Brett,

thanks for your error report. However, I am unable to repeoduce the problem using the GFF excerpt you sent. It rendered fine using the most current  version of AnnotationSketch/GenomeTools:

![out](https://cloud.githubusercontent.com/assets/27732/5798747/73399f0c-9fc2-11e4-8c48-ce81497c5bb6.png)

Which version are you using? Can you isolate a minimal example file triggering the problem for you?
The assertion message suggests that it is something related to the graph structure or feature ordering in the input file. Could you try sorting the input file first using:
```
gt gff3 -sort -tidy -retainids 4.AUGUSTUS_PREDICTION3 > 4.AUGUSTUS_PREDICTION3.sorted.gff3
```

Thanks
Sascha

**Comment by Brett-CCG** [comment]
I'm using version 1.5.1 (2014-03-22 18:04:39)

I tried sorting the GFF file as you described, but I received the same error message again.

Do you think using the latest version 1.5.5 would fix the problem? I'll try installing it and running again.

**Comment by Brett-CCG** [comment]
I don't have installation privileges on our system, so I'll have to wait for the sys admin to install.

**Comment by satta** [comment]
It's possible that the new version does not have the problem. There have been many resolved issues since then. Let's see whether running 1.5.4 or later works for you, if not I will be very happy to further debug the problem once you can provide me with a GFF3 file triggering the assertion.

**Comment by satta** [comment]
Good news: when honoring your range constraint, I also get the assertion. Before I just tried drawing the full region. I will investigate the problem, in essence I suspect it has to do with your display range beginning in the middle of a gene. So a workaround for you right now would be to draw the whole range of the gene. I will keep this ticket updated with my progress.

**Comment by Brett-CCG** [comment]
That's great. I'm still waiting for our sys admin to install the new version. But I will try as you suggest and change the range constraint.

**Comment by Brett-CCG** [comment]
I have tried by giving the coordinates of the gene from Augustus, and it works generating an image. However, I have other gene models and evidence (peptides, ESTs etc) I'm adding to the image, so I choose a range to encompass all of the evidence that was used to predict the Augustus gene model, and I end up with the same error. I used the most conservative coordinates, selecting the left most gene and the right most gene, and I get the error. The coordinates I chose were 4096968-4112761. It might be because there are other genes in the Augustus GFF which overlap with those coordinates. If I remove them, it may fix the problem, but it would be nice to be able to show in the image that there are other genes in the region, besides the one I'm focusing on. I've generated images before which have coordinates in the middle of a gene on the outer edges of the image. It may be that these genes are highly spliced, making it more difficult.
